File: python/1021_notas_e_moedas.py
N = float(input())

notas = [100, 50, 20, 10, 5, 2]
moedas = [1.00, 0.50, 0.25, 0.10, 0.05, 0.01]

centavos = int(round(N * 100))

print('NOTAS:')
for nota in notas:
    valor_nota = int(nota * 100)
    qntd_notas = centavos // valor_nota
    print(f'{qntd_notas} nota(s) de R$ {nota:.2f}')
    centavos %= valor_nota

print('MOEDAS:')
for moeda in moedas:
    valor_moeda = int(moeda * 100)
    qntd_moedas = centavos // valor_moeda
    print(f'{qntd_moedas} moeda(s) de R$ {moeda:.2f}')
    centavos %= valor_moeda

File: c/c99/1021_notas_e_moedas.c
#include <stdio.h>

int main() {
    double valor;
    int notas[] = {100, 50, 20, 10, 5, 2};
    int moedas[] = {100, 50, 25, 10, 5, 1};

    scanf("%1f", &valor);

    int valor_em_centavos = (int)(valor * 100 + 0.5);

    printf("NOTAS:\n");
    for (int i = 0; i < 6; i++) {
        int nota = notas[i];
        int quantidade = valor_em_centavos / (nota *100);
        printf("%d nota(s) de R$ %d.00\n", quantidade, nota);
        valor_em_centavos %= (nota * 100);
    }

    printf("MOEDAS:\n");
    for (int i = 0; i < 6; i++) {
        int moeda = moedas[i];
        int quantidade = valor_em_centavos / moeda;
        if (moeda >= 100)
            printf("%d moeda(s) de R$ 1.00\n", quantidade);
        else
            printf("%d moeda(s) de R$ 0.%02d\n", quantidade, moeda);
        valor_em_centavos %= moeda;
    }

    return 0;
}
